modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -145,36 +149,19 @@ private static DataType toDataType(ResolvableType type)
     {
         Class<?> rawClass = type.getRawClass();
         ResolvableType[] generics = type.getGenerics();
-        DataType resolvedDataType;
 
         if (isOperation(rawClass))
         {
-            resolvedDataType = DataType.of(OperationModel.class);
+            return DataType.of(OperationModel.class);
         }
-        else if (List.class.isAssignableFrom(rawClass))
+        else if (ArrayUtils.isEmpty(generics))
         {
-            if (!ArrayUtils.isEmpty(generics) && generics[0].getRawClass() != null)
```

Comment:
That behaviour is contained in this line -> resolvableTypes[i].resolve(Object.class);

which is called only if !ArrayUtils.isEmpty(generics)